packages/core/useCloned/index.ts

```diff
@@ -0,0 +1,43 @@
+import type { MaybeRef } from '@vueuse/shared'
+import type { Ref, WatchStopHandle } from 'vue-demi'
+import { isRef, ref, unref, watch } from 'vue-demi'
+
+export interface UseClonedOptions<T extends Record<any, any> = Record<any, any>> {
+  /**
+   * sync source only by function
+   */
+  manual?: boolean
+  /**
+   * Custom clone function should return new value for cloned data
+   */
+  cloneFunction?: (source: Partial<T>, cloned: Partial<T>) => Partial<T> | T
```

Comment:
```suggestion
  cloneFunction?: (source: T) => | T
```

And we could default it to `JSON.parse(JSON.stringify(source))`